gtk/gtk.go

```diff
@@ -8079,13 +8079,20 @@ func wrapTextTag(obj *glib.Object) *TextTag {
 	return &TextTag{obj}
 }
 
+// TextTagNew is a wrapper around gtk_text_tag_new(). If name is empty, then the
+// tag is anonymous.
 func TextTagNew(name string) (*TextTag, error) {
-	cname := C.CString(name)
-	defer C.free(unsafe.Pointer(cname))
-	c := C.gtk_text_tag_new((*C.gchar)(cname))
+	var gchar *C.gchar
+	if name != "" {
```

Comment:
never check an empty string like this. String comparison is expensive.

use `len(name) <= 0`

Reply:
"String equality comparison always begins by checking equal lengths," or so they said.

Reply:
Godbolt seems to suggest that they are indeed the same: https://godbolt.org/z/je2S2r

Reply:
For i386, seems like `<=` actually results in more instructions, but that's unfair, as a `len(str) == 0` is enough.

Reply:
Hm... I can't find that statement. But according to godbolt.org it seems to be true. My fault.  
I guess this logic applies only to `Compare` & Co. in https://golang.org/pkg/strings/. Here they state `It is usually clearer and always faster to use the built-in string comparison operators ==, <, >, and so on.`.

interesting website, didn't know that one. Thanks!  
But when I choose 386 gc (tip), then `b = len(str) <= 0` and `b = len(str) == 0` results in the same amount of instructions